game-headed/src/main/java/org/triplea/game/client/ui/javafx/TripleA.java

```diff
@@ -105,6 +114,9 @@ private void hideExitConfirm() {
   @SuppressWarnings("static-method")
   private void exit() {
     Platform.exit();
+    if (!GraphicsEnvironment.isHeadless()) {
+      SwingUtilities.invokeLater(GameRunner::exitGameIfFinished);
```

Comment:
Why is the exit conditional on the game being finished? If we're closing the window and are doing an exit, should that not just exit?

Reply:
I'm not entirely sure anymore.
I guess I was concerned that `SwingUtilities.invokeLater` would throw an exception on headless machines, not sure if that's actually the case.

Reply:
Thinking about this again:
Without this check the exit method will probably just shutdown the test suite because there aren't any visible windows.